mne_bids/write.py

```diff
@@ -1317,29 +1365,61 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
                                       suffix='T1w.nii.gz')
 
     # Check if we have necessary conditions for writing a sidecar JSON
-    if trans is not None:
-
-        # get trans and ensure it is from head to MRI
-        trans, _ = _get_trans(trans, fro='head', to='mri')
-
-        if not isinstance(raw, BaseRaw):
-            raise ValueError('`raw` must be specified if `trans` is not None')
-
-        # Prepare to write the sidecar JSON
-        # extract MEG landmarks
-        coords_dict = _extract_landmarks(raw.info['dig'])
-        meg_landmarks = np.asarray((coords_dict['LPA'],
-                                    coords_dict['NAS'],
-                                    coords_dict['RPA']))
-
-        # Transform MEG landmarks into MRI space, adjust units by * 1e3
-        mri_landmarks = apply_trans(trans, meg_landmarks, move=True) * 1e3
-
-        # Get landmarks in voxel space, using the mgh version of our T1 data
+    if trans is not None or landmarks is not None:
         t1_mgh = nib.MGHImage(t1w.dataobj, t1w.affine)
-        vox2ras_tkr = t1_mgh.header.get_vox2ras_tkr()
-        ras2vox_tkr = np.linalg.inv(vox2ras_tkr)
-        mri_landmarks = apply_trans(ras2vox_tkr, mri_landmarks)  # in vox
+
+        if landmarks is not None:
+            if raw is not None and verbose:
+                raise ValueError('Please use either `landmarks` or `raw`, ' +
+                                 'which digitization to use is ambiguous.')
+            if isinstance(landmarks, str):
+                landmarks, coord_frame = read_fiducials(landmarks)
+                landmarks = np.array([landmark['r'] for landmark in
+                                      landmarks], dtype=float)  # unpack
+            else:
+                coords_dict, coord_frame = _get_fid_coords(landmarks.dig)
+                landmarks = np.asarray((coords_dict['lpa'],
+                                        coords_dict['nasion'],
+                                        coords_dict['rpa']))
+            if coord_frame == FIFF.FIFFV_COORD_HEAD:
+                if trans is None:
+                    raise ValueError('Head space landmarks provided, ' +
+                                     '`trans` required')
+                mri_landmarks = _meg_landmarks_to_mri_landmarks(
+                    landmarks, trans)
+                mri_landmarks = _mri_landmarks_to_mri_voxels(
+                    mri_landmarks, t1_mgh)
+            elif coord_frame in (FIFF.FIFFV_MNE_COORD_MRI_VOXEL,
+                                 FIFF.FIFFV_COORD_MRI):
+                if trans is not None:
+                    raise ValueError('`trans` was provided but `landmark` ' +
+                                     'data is in mri space. Please use ' +
+                                     'only one of these.')
```

Comment:
This seems unnecessarily complicated. The Zen of Python says:

```sh
There should be one-- and preferably only one --obvious way to do it.
```

Just give users the ability to provide `landmarks` if it's in MRI space. Otherwise, throw an error. If it's in head space, you should get it from `raw.info['dig']` and `trans`

Reply:
The issue with that is that if you save landmarks in the mne coreg gui, they're in MRI RAS but not voxel space. It's pretty clearly documented in the comments above why at least voxels and RAS are needed. The head could be deleted but again as documented above, you can't save fiducials in MRI space so it's actually good to have if you do it through freesurfer. It's a complicated issue, I am trying to be zen but also it has to be useable.

Reply:
alright fair enough :)

Reply:
I originally wrote it just for MRI and thought I would be done in 5 minutes... Did not turn out to be the case lol

Reply:
The rule of thumb in estimating time is to multiply your estimate by 2 and change the units to next higher one.